unity-client/Assets/Scripts/MainScripts/DCL/Controllers/PointerEventsController.cs

```diff
@@ -41,7 +44,22 @@ public void Update()
             if (!CommonScriptableObjects.rendererState.Get() || charCamera == null) return;
 
             // We use Physics.Raycast() instead of our raycastHandler.Raycast() as that one is slower, sometimes 2x, because it fetches info we don't need here
-            if (!Physics.Raycast(GetRayFromCamera(), out hitInfo, Mathf.Infinity, PhysicsLayers.physicsCastLayerMaskWithoutCharacter))
+            // TODO: POLISH AND OPTIMIZE
+            bool didHit = Physics.Raycast(GetRayFromCamera(), out hitInfo, Mathf.Infinity, PhysicsLayers.physicsCastLayerMaskWithoutCharacter);
+            bool UIIsBlocking = false;
+
+            if (didHit)
+            {
+                GraphicRaycaster raycaster = GameObject.FindObjectOfType<Canvas>().GetComponent<GraphicRaycaster>();
```

Comment:
There's no way of having more than a single `GraphicRaycaster` in the scene? Also, this looks like a nasty approach that would give a performance hit on all detections. We should at least cache the object.

Reply:
I was just testing the aproach, this is by no means the final implementation, as the PR is still in WIP =P